### DISCUSSION TITLE
Best way to write an image using Flysystem?

### DISCUSSION THREAD

**Discussion by jcogs-design** [avalanche123/Imagine#854] [Q&A] [answered]
Am trying to integrate Image with a Flysystem implementation.

Wondering about the best way to write images out using the [Flysystem](https://flysystem.thephpleague.com/docs/) "write" method.

For that to work the image needs to be encoded in whatever image format is required before it is written.

One option would be to write the image out to a local disk first and then copy that local file to the remote store - but wondering if there is a way to encode the image without writing to disk first? Would seem more elegant if so.

**Comment by ausi** [comment, marked as answer]
You can use `ImageInterface::get()` that returns the binary data of the image that you can then write to the Flysystem:

```php
$filesystem->write($path, $image->get($format, $options), $config);
```

**Reply by jcogs-design** [reply]
That's great - thanks for the speedy reply.
